misc/protobuf-ballerina/src/main/java/org/ballerinalang/protobuf/cmd/GrpcCmd.java

```diff
@@ -99,52 +102,69 @@
+
+        if (protoPath == null || !protoPath.toLowerCase(Locale.ENGLISH).endsWith(PROTO_SUFFIX)) {
+            String errorMessage = "Invalid proto file path. Please input valid proto file location.";
             outStream.println(errorMessage);
             throw new BalGenToolException(errorMessage);
         }
-        protoPath = protoFile.getAbsolutePath();
+
+        if (!Files.isReadable(Paths.get(protoPath))) {
+            String errorMessage = "Provided service proto file is not readable. Please input valid proto file " +
+                    "location.";
+            outStream.println(errorMessage);
+            throw new BalGenToolException(errorMessage);
+        }
+
         if (helpFlag) {
-            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(parentCmdParser, "build");
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo("build");
             outStream.println(commandUsageInfo);
             return;
         }
+
         try {
             downloadProtocexe();
         } catch (BalGenToolException e) {
             LOG.error("Error while generating protoc executable. ", e);
             throw new BalGenToolException("Error while generating protoc executable. ", e);
         }
-        ClassLoader classLoader = this.getClass().getClassLoader();
-        String descriptorPath = BalGenerationConstants.META_LOCATION + getProtoFileName() + "-descriptor.desc";
-        File descFile = generateDescTempFolder(descriptorPath);
+
+        File descFile = generateTempDirectory();
         StringBuilder msg = new StringBuilder();
         LOG.debug("Initializing the ballerina code generation.");
-        List<String> protoFiles = readProperties(classLoader);
-        for (String file : protoFiles) {
-            try {
-                exportResource(file, classLoader);
-            } catch (Exception e) {
-                msg.append("Error extracting resource file ").append(file).append(NEW_LINE_CHARACTER);
-                outStream.println(msg.toString());
-                LOG.error("Error exacting resource file " + file, e);
+
+        byte[] root;
+        List<byte[]> dependant;
+
+        try {
+            ClassLoader classLoader = this.getClass().getClassLoader();
+            List<String> protoFiles = readProperties(classLoader);
+            for (String file : protoFiles) {
+                try {
+                    exportResource(file, classLoader);
+                } catch (Exception e) {
+                    msg.append("Error extracting resource file ").append(file).append(NEW_LINE_CHARACTER);
+                    outStream.println(msg.toString());
+                    LOG.error("Error exacting resource file " + file, e);
+                }
             }
+            msg.append("Successfully generated initial files.").append(NEW_LINE_CHARACTER);
+            root = BalFileGenerationUtils.getProtoByteArray(this.exePath, this.protoPath, descFile
```

Comment:
Shall we reformat this line?

Reply:
Fixed in f2bb0753488e0c357b61ea5a9d3639e43bdf9bfe